src/devices/cddadevice.h

```diff
@@ -18,31 +18,43 @@
 #ifndef CDDADEVICE_H
 #define CDDADEVICE_H
 
-#include <QMutex>
+#include <QTimer>
 
 // These must come after Qt includes (issue 3247)
 #include <cdio/cdio.h>
-#include <gst/audio/gstaudiocdsrc.h>
 
 #include "cddasongloader.h"
 #include "connecteddevice.h"
 #include "core/song.h"
-#include "musicbrainz/musicbrainzclient.h"
+
+class QUrl;
 
 class CddaDevice : public ConnectedDevice {
   Q_OBJECT
 
  public:
+  // Initializes the CddaDevice but does not open a handle to the device.
+  // Using code MUST call Init() after construction before calling and
+  // check that it returns true any other methods, to ensure that a valid
+  // device handle was correctly opened. Using class methods without
+  // a valid device handle is undefined behavior and might result in crashes.
   Q_INVOKABLE CddaDevice(const QUrl& url, DeviceLister* lister,
                          const QString& unique_id, DeviceManager* manager,
-                         Application* app, int database_id, bool first_time);
+                         Application* app, int database_id, bool first_time,
+                         bool watch_for_disc_changes = true);
```

Comment:
Is there (or will there be) a use case where this isn't true?

Reply:
I guess not, can probably remove that from the constructor.

Reply:
If you're planning on doing that, you could also start the timer from Init() and you wouldn't have to worry about the timer being called before initialization.

Reply:
done